code/game/machinery/washing_machine.dm

```diff
@@ -64,14 +69,23 @@
 		if(wash_color)
 			for(var/obj/item/clothing/I in contents)
 				I.color = wash_color //Simply recolor the items.
+				I.desc = I.desc + (length(I.desc) > 0 ? " " : "") + "It looks soaked in [wash_color] tincture."
 
 				var/list/colors = list("light brown", "brown", "cyan", "fingerless", "combat", "tactical", "yellowgreen", "darkred", "lightred", "maroon", "red", "orange", "rainbow", "lightgreen", "green", "lightpurple", "purple", "gold", "darkblue", "lightblue", "aqua", "blue", "yellow", "black", "grey", "gray", "white", "latex", "nitrile", "budget insulated", "insulated", "captain's")
 			
 				for(var/old_color in colors)
 					if(findtext(I.name, old_color))
 						I.name = n_replace(I.name, old_color, wash_color)
-						I.desc = I.desc + (length(I.desc) > 0 ? " " : "") + "It looks soaked in [wash_color] tincture."
 						break
+
+				for(var/obj/item/clothing/J in I.contents)
+					J.color = wash_color
+					J.desc = J.desc + (length(J.desc) > 0 ? " " : "") + "It looks soaked in [wash_color] tincture."
+
+					for(var/old_color in colors)
+						if(findtext(J.name, old_color))
+							J.name = n_replace(J.name, old_color, wash_color)
+							break
```

Comment:
I dunno, maybe there's  better way to do this than copy-pasting code?
Perhaps, you could have double for-loops checking contents of both src and I, add all found objects to the var/list/L and then for(var/obj/item/clothing/I in L you'd execute the code?
Though that would mean THREE for loops as opposed to two. Fuck.